shared/login/relogin/container.tsx

```diff
@@ -32,57 +23,65 @@ type Props = {
   users: Array<ConfigTypes.ConfiguredAccount>
 }
 
-class LoginWrapper extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props)
-    this.state = {
-      inputKey: 1,
-      password: '',
-      selectedUser: props.selectedUser,
-      showTyping: false,
-    }
-  }
+const LoginWrapper = (props: Props) => {
+  const [password, setPassword] = React.useState('')
+  const [selectedUser, setSelectedUser] = React.useState(props.selectedUser)
+  const [showTyping, setShowTyping] = React.useState(false)
 
-  _selectedUserChange = (selectedUser: string) => {
-    this.setState({selectedUser})
-    if (this.props.loggedInMap.get(selectedUser)) {
-      this.props.onLogin(selectedUser, '')
-    }
-  }
+  const prevPassword = Container.usePrevious(password)
+  const prevError = Container.usePrevious(props.error)
 
-  componentDidUpdate(prevProps: Props) {
-    // Clear the password when there's an input error.
-    if (this.props.inputError !== prevProps.inputError) {
-      this.setState(p => ({inputKey: p.inputKey + 1, password: ''}))
+  const dispatch = Container.useDispatch()
+
+  const onSubmit = React.useCallback(() => {
+    props.onLogin(selectedUser, password)
+  }, [selectedUser, password, props])
+
+  const selectedUserChange = React.useCallback(
+    user => {
+      dispatch(LoginGen.createLoginError({error: null}))
+      setPassword('')
+      setSelectedUser(user)
+      if (props.loggedInMap.get(user)) {
+        props.onLogin(user, '')
+      }
+    },
+    [dispatch, setPassword, setSelectedUser, props]
```

Comment:
same here

Reply:
lint yells at me if I don't include props, but I can destructure.

Reply:
You can pass props.onlogin or whatever

Reply:
> react-hooks/exhaustive-deps: React Hook React.useCallback has a missing dependency: 'props'. Either include it or remove the dependency array. However 'props' will change when *any* prop changes, so the preferred fix is to destructure the 'props' object outside of the useCallback call and refer to those specific props inside React.useCallback.